dissect/target/plugins/os/unix/log/audit.py

```diff
@@ -1,60 +1,89 @@
-import datetime
 import re
+from pathlib import Path
+from typing import Iterator
 
+from dissect.util.ts import from_unix
+
+from dissect.target.helpers.fsutil import basename, open_decompress
 from dissect.target.helpers.record import TargetRecordDescriptor
-from dissect.target.plugin import Plugin, export
+from dissect.target.plugin import Plugin, export, internal
 
 AuditRecord = TargetRecordDescriptor(
     "linux/log/audit",
     [
         ("datetime", "ts"),
-        ("string", "type"),
-        ("string", "id"),
-        ("string", "msg"),
-        ("string", "auditlog"),
+        ("string", "audit_type"),
+        ("varint", "audit_id"),
+        ("string", "message"),
+        ("path", "source"),
     ],
 )
 
+AUDIT_REGEX = re.compile(r"^type=(?P<audit_type>.*) msg=audit\((?P<ts>.*):(?P<audit_id>.*)\): (?P<message>.*)$")
+
 
 class AuditPlugin(Plugin):
-    def check_compatible(self):
-        path = self.target.fs.path("/var/log/audit/")
+    def __init__(self, target):
+        super().__init__(target)
+        self.log_paths = self.get_log_paths()
+
+    def check_compatible(self) -> bool:
+        return len(self.log_paths) > 0
+
+    @internal
+    def get_log_paths(self) -> list[Path]:
+        log_paths = []
+
+        log_paths.extend(self.target.fs.path("/var/log/audit").glob("audit.log*"))
 
-        if path.exists():
-            return True
-        else:
-            return False
+        default_config = "/etc/audit/auditd.conf"
```

Comment:
Can you add a unit test for parsing the config btw?

Reply:
Added in [ea42c88](https://github.com/fox-it/dissect.target/pull/173/commits/ea42c884e249edbb360188c4ee0f13e0da58b82b).